### ISSUE TITLE
[FEATURE_REQUEST] Add check for valid Port Names / Length

### ISSUE THREAD

**Issue by mstoetzer** [stackrox/kube-linter#124]
**Description of the problem/feature request**
Kube-Linter should check for valid port names.

**Description of the existing behavior vs. expected behavior**
Kubectl checks for port names and displays an error if they are to long.

```
Error from server (Invalid): Service "foobar-event-management" is invalid: spec.ports\[0\].targetPort: Invalid value: "foobar-event-management": must be no more than 15 characters
```

Could this check be also added to kube-linter somehow?

**Comment by viswajithiii** [comment]
Thanks @mstoetzer! We can definitely look into this, and this is something we would like to add to KubeLinter. Can you paste the whole failing YAML file, just in case we need it for debugging?

**Comment by mstoetzer** [comment]
Please see the following `here document`:

```yaml
cat <<EOF | kubectl apply -f - --dry-run=server
apiVersion: apps/v1
kind: Deployment
metadata:
  name: some-app
  labels:
    app: app
spec:
  replicas: 1
  selector:
    matchLabels:
      app: app
  template:
    metadata:
      labels:
        app: app
    spec:
      containers:
        - name: app
          image: does-not-matter
          ports:
            - name: port-that-works
              containerPort: 8080
            - name: port-that-is-too-long
              containerPort: 9000
EOF
```

This produces the following output:

```
The Deployment "some-app" is invalid: spec.template.spec.containers[0].ports[1].name: Invalid value: "port-that-is-too-long": must be no more than 15 characters
```

**Comment by mstoetzer** [comment]
This also applies for services:

```
cat <<EOF | kubectl apply -f - --dry-run=server
apiVersion: v1
kind: Service
metadata:
  name: some-service
spec:
  ports:
    - port: 80
      targetPort: port-that-is-too-long
      protocol: TCP
      name: http
  selector:
    app: app
EOF
```

Which produces:

```
The Service "some-service" is invalid: spec.ports[0].targetPort: Invalid value: "port-that-is-too-long": must be no more than 15 characters
```

**Comment by misberner** [comment]
@viswajithiii I would broaden the scope of this feature request to do full Kubernetes schema validation (cf. what I did for helmtest).

**Comment by mstoetzer** [comment]
@viswajithiii Would it be the correct approach to create a check template for ports (or port names?)? A custom check could be defined using a regex, that limits a port name to 15 characters.

**Comment by viswajithiii** [comment]
@mstoetzer That would be the quickest approach. The more long-term approach would be to implement #36, which would involve checking all objects against the schema -- that would catch this and other similar problems. However, I'm okay with the short-term approach in the meantime as well, if you were planning to send a PR with that!
